app_test.go

```diff
@@ -2287,3 +2291,66 @@ func TestSetupInitializesOnlyNilWriters(t *testing.T) {
 		t.Errorf("expected a.Writer to be os.Stdout")
 	}
 }
+
+func TestFlagAction(t *testing.T) {
+	r := []string{}
+	actionFunc := func(c *Context, s string) error {
+		r = append(r, s)
+		return nil
+	}
+
+	app := &App{
+		Name:   "command",
+		Writer: io.Discard,
+		Flags:  []Flag{&StringFlag{Name: "flag", Action: actionFunc}},
```

Comment:
Can you add some more tests which use other types of flags ? codecov is complaining that runAction on other flags dont have any test coverage

Reply:
No problem,  I'll fill it in.

Reply:
Updated.